src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java

```diff
@@ -97,38 +103,49 @@ public void setup() {
         customShelfService.save(customShelf);
 
         Assumptions.assumeTrue(predefinedShelfService != null);
-        bookForm = createBookForm(true);
-
+        bookForm = createBookForm(READ, true);
     }
 
-    private BookForm createBookForm(boolean isInSeries) {
+    private BookForm createBookForm(ShelfName shelf, boolean isInSeries) {
         BookForm bookForm = new BookForm(predefinedShelfService, customShelfService);
-        bookForm.setBook(createBook(predefinedShelfService, isInSeries));
+        readShelf = predefinedShelfService.findAll().get(2);
+        bookForm.setBook(createBook(shelf, isInSeries));
         return bookForm;
-
     }
 
-    private Book createBook(PredefinedShelfService predefinedShelfService, boolean isInSeries) {
+    private Book createBook(ShelfName shelfName, boolean isInSeries) {
+        // values that are present in all forms
         Author author = new Author(firstName, lastName);
-        readShelf = predefinedShelfService.findAll().get(2);
-        Book book = new Book(bookTitle, author, readShelf);
-
+        PredefinedShelf shelf = predefinedShelfService.findAll(shelfName).get(0);
```

Comment:
Could you replace `predefinedShelfService.findAll(shelfName).get(0)` with the `findPredefinedShelf()` method in `PredefinedShelfUtils`?

Reply:
There is the same "Non-static method cannot be referenced from a static context" error.

Reply:
I've addressed this in that suggestion. With the global variable for the instance to `PredefinedShelfUtils`, you should be able to get this to work.

Reply:
Yes, that helped, thank you. Now its done.